File: src/main/java/com/android/helper/base/title/HttpCallBackListener.java
package com.android.helper.base.title;

import org.jetbrains.annotations.NotNull;

/**
 * @author : 流星
 * @CreateDate: 2022/1/1-18:23
 * @Description:
 */
public interface HttpCallBackListener<T> {

    void onHttpStart();

    void onHttpSuccess(@NotNull T t);

    void onHttpError(@NotNull Throwable e);

    void onHttpComplete();
}

File: src/main/java/com/android/helper/httpclient/kotlin/listener/HttpCallBackListener.kt
package com.android.helper.httpclient.kotlin.listener

/**
 * @author : 流星
 * @CreateDate: 2023/3/15-17:31
 * @Description:
 */
interface HttpCallBackListener<T> {
    fun onStart()
    fun onSuccess(t: T)
    fun onFailure(exception: Throwable)
    fun onCompletion()
}